### PR TITLE
Admin destroy

### PR DESCRIPTION
**Pull request by ehuelsmann**
Follow-up to @ylavoie's PR #5384 which has been marked failed for too long now.

### CODE REVIEW

File: lib/LedgerSMB/Admin/Command/destroy.pm [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+use Syntax::Keyword::Try;
+
+use LedgerSMB::Admin::Command;
+use LedgerSMB::Database;
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+my $logger;
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/destroy.pm:22]: @ylavoie why not simply declare the logger internal to the `run()` method?

File: lib/LedgerSMB/Admin/Command/destroy.pm [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    # Scan databases which could be using the same profile
+    my $sth =
+        $dbh->prepare('SELECT datname FROM pg_database WHERE NOT datistemplate');
+
+    $sth->execute
+        or die $dbh->errstr;
+
+    my @lsmb_databases;
+    while (my $database = $sth->fetchrow) {
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/destroy.pm:59]: @ylavoie why first put them in a list and then ...

File: lib/LedgerSMB/Admin/Command/destroy.pm [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        push @lsmb_databases, $database;
+    }
+    $sth->finish;
+    $dbh->disconnect;
+
+    my $role_prefix_usage = 0;
+    for my $database (@lsmb_databases) {
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/destroy.pm:66]: ... iterate over *that* list to connect to the databases?

I'm not seeing which other role the list plays in this function, so I don't see its value.

File: lib/LedgerSMB/Admin/Command/destroy.pm [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                    connect_data => $connect_database,
+                );
+            my $database_role_prefix = _get_role_prefix($db);
+            if ( defined $database_role_prefix && $role_prefix eq $database_role_prefix ) {
+                $role_prefix_usage++;
+                $logger->debug("'$role_prefix' used by $database");
+            }
+        }
+        catch ($e) {
+            $logger->error("ERROR: $e");
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/destroy.pm:83]: We need to prefix the error message with "ERROR:" even when already using the `error` logger???

**Review by ehuelsmann — COMMENTED** [review]
Some remarks with respect to the code.

File: lib/LedgerSMB/Admin/Command/destroy.pm (hunk 1 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/destroy.pm:22]: It is used in other functions

File: lib/LedgerSMB/Admin/Command/destroy.pm (hunk 4 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/destroy.pm:83]: Redundant. Fixed

**Comment by ehuelsmann** [comment]
Closing this one as comments have been merged into #5384

**Pull request closed without merging**